### PR TITLE
[New Rule] Azure Automation Runbook Deleted

### LINKED ISSUE #117: [New Rule] Azure Automation Runbook Deleted
## Description

An adversary may delete an Azure Automation runbook in order to disrupt automated operations that may be configured to execute on a scheduled basis.

```
event.module:azure and event.dataset:azure.activitylogs and event.category:Administrative and azure.activitylogs.operation_name:(MICROSOFT.AUTOMATION/AUTOMATIONACCOUNTS/RUNBOOKS/DELETE) and event.outcome: "Success"
```

## Required Info

- **Eventing Sources:**

`filebeat-*`

- **Target Operating Systems:**

- **Platforms**

Azure

- **Target ECS Version:** `1.5.0`
- **New fields required in ECS for this?** No
- **Related issues or PRs** None

## Optional Info
- **References:**
  - https://powerzure.readthedocs.io/en/latest/Functions/operational.html#create-backdoor
  - https://github.com/hausec/PowerZure
  - https://posts.specterops.io/attacking-azure-azure-ad-and-introducing-powerzure-ca70b330511a
  - https://azure.microsoft.com/en-in/blog/azure-automation-runbook-management/

## Example Data

image

### PR DESCRIPTION
**Pull request by threat-punter**
## Issues

Resolves #117 

## Summary
Identifies when an Azure Automation runbook is deleted. An adversary may delete an Azure Automation runbook in order to disrupt atheir target's automated business operations or to remove a malicious runbook that was used for persistence.

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/azure/impact_azure_automation_runbook_deleted.toml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+creation_date = "2020/09/01"
+ecs_version = ["1.6.0"]
+maturity = "production"
+updated_date = "2020/09/01"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies when an Azure Automation runbook is deleted. An adversary may delete an Azure Automation runbook in order to
+disrupt atheir target's automated business operations or to remove a malicious runbook that was used for persistence.
```
Comment (brokensound77) [inline rules/azure/impact_azure_automation_runbook_deleted.toml:11]:
```suggestion
disrupt their target's automated business operations or to remove a malicious runbook that was used for persistence.
```

File: rules/azure/impact_azure_automation_runbook_deleted.toml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    "https://azure.microsoft.com/en-in/blog/azure-automation-runbook-management/",
+]
+risk_score = 21
+rule_id = "8ddab73b-3d15-4e5d-9413-47f05553c1d7"
+severity = "low"
+tags = ["Elastic", "Azure", "Continuous Monitoring", "SecOps", "Configuration Audit"]
+type = "query"
+
+query = '''
+event.module:azure and event.dataset:azure.activitylogs and event.category:Administrative and azure.activitylogs.operation_name:(MICROSOFT.AUTOMATION/AUTOMATIONACCOUNTS/RUNBOOKS/DELETE) and event.outcome:Success
```
Comment (brokensound77) [inline rules/azure/impact_azure_automation_runbook_deleted.toml:32]: nit:

```suggestion
event.module:azure and event.dataset:azure.activitylogs and event.category:Administrative and azure.activitylogs.operation_name:MICROSOFT.AUTOMATION/AUTOMATIONACCOUNTS/RUNBOOKS/DELETE and event.outcome:Success
```

**Review by peasead — APPROVED** [review]
LGTM w/ @brokensound77's suggestions.

**Comment by threat-punter** [comment]
Cheers y'all

**Pull request merged**